### PR TITLE
Chore/auto generate mock providers

### LINKED ISSUE #1537: [Housekeeping] Auto generate mock providers
**Short description of housekeeping request**
We should replace the duplicated and hardcoded mock providers in KirbyTestingModule (jasmine+jest) with auto generated ones to ensure the mocked providers are always in sync with the actual providers (exported classes + public methods).

**Alternatives**
Manually keep them in sync or let them drift apart... ⛵ 

**Additional context**
We already do this for components 👌🏻 

### Kick Off

* [X] Ensure this issue has been `Tech refined` with @kirbydesign/kirby-guild and is updated with a clear implementation description
  _This issue should be in the [Ready to do](https://github.com/kirbydesign/designsystem/projects/1#column-4590936) column of the [Kirby kan-ban board](https://github.com/kirbydesign/designsystem/projects/1) before starting implementation_
* [X] Assign yourself to this issue and move it to the [In progress](https://github.com/kirbydesign/designsystem/projects/1#column-4590937) column of the [Kirby kan-ban board](https://github.com/kirbydesign/designsystem/projects/1)

### Code

* [X] Create Feature Branch from [master branch](https://github.com/kirbydesign/designsystem/tree/master)
* [ ] ~~Implement/update unit tests~~
* [X] Create a draft implementation and push to Github
* [ ] ~~Ask a member of @kirbydesign/kirby-guild for a WIP review by creating a draft Pull Request~~

### Code Review

* [x] Open a pull request (or mark the existing draft PR as `Ready for review`) and ask @kirbydesign/kirby-guild for a review
  _Remember to add `closes #issueno` to the description of the PR._
* [ ] Once approved, merge feature branch/PR to master

:tada: Celebrate

### PR DESCRIPTION
**Pull request by jakobe**
This PR closes #1537 

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] ~~Tests for the changes have been added (for bug fixes / features)~~
- [ ] ~~Docs have been added / updated (for bug fixes / features)~~

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Enhancement (to existing content)
- [ ] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] cookbook application / infrastructure changes
- [ ] Other... Please describe:

## Which issue documents the current behaviour?

Issue Number: #1537 

## What is the new behavior?
Auto-generating mocked providers instead of hardcoded / duplicated ones.

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: tools/generate-mocks/generate-mocks.ts [hunk 1]
```diff
@@ -14,20 +15,30 @@ type ComponentMetaData = {
...
     await this.traverseFolder(
       inputPath,
       outputPathNormalized,
-      subPath,
       classMap,
-      exportedComponents
+      exportedTypes,
+      exportedProviders
     );
+    // console.log('exportedProviders:', exportedProviders);
```
Comment (flemcito) [inline tools/generate-mocks/generate-mocks.ts:38]: log statement

File: tools/generate-mocks/generate-mocks.ts [hunk 2]
```diff
@@ -92,36 +140,72 @@ export const MOCK_COMPONENTS = [
...
     const folderContent = readdirSync(folderpath);
     for (const fileOrFolder of folderContent) {
       const fullPath = path.join(folderpath, fileOrFolder);
       const ent = statSync(fullPath);
       if (ent.isDirectory()) {
-        await this.traverseFolder(fullPath, outputPath, subPath, classMap, exportedComponents);
+        await this.traverseFolder(fullPath, outputPath, classMap, exportedTypes, exportedProviders);
       } else {
         if (fileOrFolder.endsWith('.component.ts')) {
           // console.log('Rendering mock for: ', fullPath);
```
Comment (flemcito) [inline tools/generate-mocks/generate-mocks.ts:155]: log statement

File: tools/generate-mocks/generate-mocks.ts [hunk 3]
```diff
@@ -92,36 +140,72 @@ export const MOCK_COMPONENTS = [
...
+
+  private renderMock(fileName: string, newFilename: string, exportedTypes: string[]) {
     const components = this.generateMetaData(fileName);
-    if (!components.find((metaData) => metaData.decorator)) {
-      // Nothing to generate:
-      return;
-    }
-    if (!components.some((metaData) => exportedComponents.indexOf(metaData.className) > -1)) {
+    if (
+      !components.some(
```
Comment (flemcito) [inline tools/generate-mocks/generate-mocks.ts:205]: consider extracting this expression to a well-named method

File: tools/generate-mocks/generate-mocks.ts [hunk 4]
```diff
@@ -192,7 +276,10 @@ export class ${mockClassName} {${propertiesString}${methodsString}}
 
   private getImports(components: ComponentMetaData[]): string[] {
     const importStatements = [];
-    const angularCoreImports = ['forwardRef'];
+    const angularCoreImports: string[] = [];
+    if (components.some((metaData) => ['Component', 'Directive'].includes(metaData.decorator))) {
```
Comment (flemcito) [inline tools/generate-mocks/generate-mocks.ts:280]: Consider assigning expression to well-named const and use that in the `if`

**Review by flemcito — CHANGES_REQUESTED** [review]
Looks great, absolutely minor comments
Really nice to make this automated 
giphub

File: tools/generate-mocks/generate-mocks.ts (hunk 4 above)
Comment (jakobe) [reply tools/generate-mocks/generate-mocks.ts:297]: Fixed 👍🏻

File: tools/generate-mocks/generate-mocks.ts (hunk 3 above)
Comment (jakobe) [reply tools/generate-mocks/generate-mocks.ts:222]: Fixed 👍🏻

**Review by flemcito — APPROVED** [review]

**Pull request merged**